packages/extension/src/newtab/App.tsx

```diff
@@ -114,6 +115,7 @@ function InternalApp(): ReactElement {
   return (
     <DndContextProvider>
       <LazyModalElement />
+      <ReputationPrivilegesModalTrigger />
```

Comment:
As mentioned in RFC the hook would cause re-renders at the top level, like this you limit the logic and state dependencies to this component without the whole app re-rendering when conditions change.

Reply:
Is there a place where we can introduce this app wide? (so avoiding the 2 implementations?)

Reply:
by two implementations you mean in extension and app roots?

Reply:
Yes correct, if possible would want it only in 1 place so it's more maintainable

Reply:
I'll take a look.

Reply:
Moved to main layout 744a22c89b5d92bbe255b134c371812dde39119d